src/interfaces/IAccountStateValidator.sol

```diff
@@ -1,18 +1,23 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.23;
 
+/// @dev Magic value returned by validateAccountState on success
+bytes4 constant VALIDATION_SUCCESS = 0x61014884; // bytes4(keccak256("validateAccountState(address,address)"))
+
 /// @title IAccountStateValidator
-///
-/// @notice Interface for account-specific validation logic
-///
+/// @notice Interface for account-specific validation logi
 /// @dev This interface is used to validate the state of a account after an upgrade
-///
 /// @author Coinbase (https://github.com/base/eip-7702-proxy)
 interface IAccountStateValidator {
+    /// @notice Error thrown when the implementation provided to `validateAccountState` is not the validator's expected implementation
+    error InvalidImplementation(address expected, address actual);
```

Comment:
Thinking we should add a view function to this interface for `supportedImplementation() returns (address)`